c2rust-analyze/src/util.rs

```diff
@@ -2,11 +2,40 @@ use crate::labeled_ty::LabeledTy;
 use rustc_hir::def::DefKind;
 use rustc_hir::def_id::DefId;
 use rustc_middle::mir::{
-    Field, Local, Mutability, Operand, PlaceElem, PlaceRef, ProjectionElem, Rvalue,
+    Field, Local, Mutability, Operand, Place, PlaceElem, PlaceRef, ProjectionElem, Rvalue,
 };
 use rustc_middle::ty::{AdtDef, DefIdTree, SubstsRef, Ty, TyCtxt, TyKind, UintTy};
+use std::collections::HashMap;
 use std::fmt::Debug;
 
+/// A mapping for substituting [`Place`]s adhering to the
+/// following pattern
+/// ```mir
+/// _1 = malloc(...);
+/// _2 = _1 as *mut T;
+/// ```
+///
+/// In this case, `_1` would be mapped to `_2`, which is indicative
+/// of the amended statement:
+/// ```
+/// _2 = malloc(...);
+/// ```
+#[derive(Default)]
+pub struct CVoidCasts<'tcx>(pub HashMap<Place<'tcx>, Place<'tcx>>);
+
+impl<'tcx> CVoidCasts<'tcx> {
+    /// Checks if the casted-to or casted-from value
+    /// is a c_void pointer.
```

Comment:
```suggestion
    /// Checks if the casted-to or casted-from value is a [`*c_void`](core::ffi::c_void).
```

Reply:
I recall you mentioning we should avoid specific `c_void` instances. Is that a concern here?

Reply:
I don't think so. This one is stable now and is just for docs anyways.